Adyen/UI/View Controllers/AddressLookup/Search/AddressLookupSearchViewController+ViewModel.swift

```diff
@@ -0,0 +1,142 @@
+            showFormHandler: @escaping (PostalAddress?) -> Void,
+            switchToManualEntryHandler: @escaping () -> Void,
+            cancellationHandler: @escaping () -> Void
+        ) {
+            self.localizationParameters = localizationParameters
+            self.style = style
+            
+            self.lookupProvider = lookupProvider
+            
+            self.presentationHandler = presentationHandler
+            self.showFormHandler = showFormHandler
+            self.switchToManualEntryHandler = switchToManualEntryHandler
+            self.cancellationHandler = cancellationHandler
+        }
+        
+        internal func handleCancelTapped() {
+            cancellationHandler()
+        }
+        
+        internal func handleSwitchToManualEntry() {
+            switchToManualEntryHandler()
+        }
+        
+        internal func handleLookUp(searchTerm: String, resultHandler: @escaping ([ListItem]) -> Void) {
+            
+            lookupProvider?.lookUp(searchTerm: searchTerm) { result in
+                
+                let listItems = result.compactMap(listItem(for:))
+                guard !listItems.isEmpty else { return resultHandler([]) }
+                resultHandler([manualEntryListItem] + listItems)
+            }
+        }
+        
+        internal func handleDidSelect(item: ListItem, addressModel: LookupAddressModel) {
+            guard let lookupProvider else { return }
+            
+            item.startLoading()
+            
+            lookupProvider.complete(incompleteAddress: addressModel) { [weak item] result in
+                item?.stopLoading()
+                
+                switch result {
+                case let .success(address):
+                    self.handleShowForm(with: address)
+                case let .failure(error):
+                    self.handleShowError(error)
+                }
+            }
+        }
+        
+        internal func handleShowForm(with address: PostalAddress?) {
+            showFormHandler(address)
+        }
+        
+        internal func handleShowError(_ error: Error) {
+            let alert = UIAlertController(
+                title: localizedString(.errorTitle, localizationParameters),
+                message: error.localizedDescription,
+                preferredStyle: .alert
+            )
+            alert.addAction(.init(
+                title: localizedString(.dismissButton, localizationParameters),
+                style: .default
+            )
+            )
+            presentationHandler(alert)
+        }
+    }
+}
+
+private extension AddressLookupSearchViewController.ViewModel {
+    
+    var manualEntryListItem: ListItem {
+        
+        .init(
+            title: localizedString(.addressLookupSearchManualEntryItemTitle, localizationParameters),
+            style: style.manualEntryListItem
+        ) {
+            self.switchToManualEntryHandler()
```

Comment:
no need for weak self?

Reply:
Good catch! Not needed here as the ViewModel is a struct 👌